velero-plugin-for-microsoft-azure/object_store.go

```diff
@@ -244,7 +249,52 @@ func (o *ObjectStore) PutObject(bucket, key string, body io.Reader) error {
 		return err
 	}
 
-	return errors.WithStack(blob.CreateBlockBlobFromReader(body, nil))
+	// Azure requires a blob/object to be chunked if it's larger than 256MB. Since we
+	// don't know ahead of time if the body is over this limit or not, and it would
+	// require reading the entire object into memory to determine the size, we use the
+	// chunking approach for all objects.
+
+	var (
+		// blocks must be less than/equal to 100MB in size
+		// ref. https://docs.microsoft.com/en-us/rest/api/storageservices/put-block#uri-parameters
+		block    = make([]byte, 100*1024*1024)
```

Comment:
we may want to make the block size (a) smaller by default to minimize memory usage, and (b) configurable.

Reply:
Fair, although 100MiB of ECC memory costs something like 50 cents these days (not per month- that's the purchase price).